p2p/swarm.go

```diff
@@ -299,36 +279,16 @@ func (s *swarm) sendMessageImpl(peerPubKey string, protocol string, payload serv
 		return fmt.Errorf("protocolMsg has unexpected type %T", x)
 	}
 
-	err = message.SignMessage(s.lNode.PrivateKey(), protomessage)
-	if err != nil {
-		return err
-	}
-
 	data, err := proto.Marshal(protomessage)
 	if err != nil {
 		return fmt.Errorf("failed to encode signed message err: %v", err)
 	}
 
-	session.EncryptGuard().Lock()
-
-	// messages must be sent in the same order as the order that the messages were encrypted because the iv used to encrypt
-	// (and therefore decrypt) is the last encrypted block of the previous message that were encrypted
-	final, err := message.PrepareMessage(session, data)
-
-	if err != nil {
-		session.EncryptGuard().Unlock()
-		// since it is possible that the encryption succeeded and the iv was modified for the next message, we must close the connection otherwise
-		// the missing message will prevent the receiver from decrypting any future message
-		s.lNode.Logger.Error("prepare message failed, closing the connection")
-		conn.Close()
-		e := fmt.Errorf("aborting send - failed to encrypt protocolMsg: %v", err)
-		return e
-	}
+	final := session.SealMessage(data)
```

Comment:
usually this was locked with the send code. so we'll only send in the right order of encryption. how is this solved here ?

Reply:
https://github.com/spacemeshos/go-spacemesh/issues/463